orx-osc/src/main/kotlin/OSC.kt

```diff
@@ -0,0 +1,71 @@
+package org.openrndr.extra.osc
+
+import com.illposed.osc.OSCMessage
+import com.illposed.osc.OSCMessageListener
+import com.illposed.osc.messageselector.OSCPatternAddressMessageSelector
+import com.illposed.osc.transport.udp.OSCPort
+import com.illposed.osc.transport.udp.OSCPortIn
+import com.illposed.osc.transport.udp.OSCPortOut
+import mu.KotlinLogging
+import java.net.InetAddress
+import java.net.PortUnreachableException
+
+private typealias OSCListener = Pair<OSCPatternAddressMessageSelector, OSCMessageListener>
+
+private val logger = KotlinLogging.logger {}
+
+class OSC (
+        val address: InetAddress = InetAddress.getLocalHost(),
+        val portIn: Int = OSCPort.DEFAULT_SC_OSC_PORT,
+        val portOut: Int = portIn
+) {
+    private val receiver: OSCPortIn = OSCPortIn(portIn)
+    private val sender: OSCPortOut = OSCPortOut(address, portOut)
+    private val listeners: MutableMap<String, OSCListener> = mutableMapOf()
+
+    fun <T> send(channel: String, vararg message: T) {
+        if (!sender.isConnected) sender.connect()
+
+        val msg = OSCMessage(channel, message.toList())
+
+        try {
+            sender.send(msg)
+        } catch (ex: Exception) {
```

Comment:
As far as I understand catching any exception here might be not what we want. If any other exception would happen, it will be just swallowed silently. Why not just: 
```
        try {
            sender.send(msg)
        } catch (ex: PortUnreachableException) {
            logger.error(ex) { "Error: Could not connect to OUT port" }
        } catch (ex: IllegalStateException) {
            logger.error(ex) { "Error: Couldn't send message to channel: $channel" }
        }
```

Reply:
@morisil that's a fair point. @ricardomatias can you adjust the exception handling as suggested?

Reply:
Yes I can.